extensions/amp-analytics/0.1/vendors/snowplow_v2.json

```diff
@@ -10,23 +10,24 @@
   "cookies": {
     "enabled": true,
     "_sp_duid": {
-      "value": "$IF($IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $IF(COOKIE(_sp_duid), COOKIE(_sp_duid), LINKER_PARAM(linker,_sp_duid)), $SUBSTR(QUERY_PARAM(_sp),0,36))"
+      "value": "$IF($SUBSTR(QUERY_PARAM(_sp),0,36), $SUBSTR(QUERY_PARAM(_sp),0,36), $IF(LINKER_PARAM(sp_amp_linker,_sp_duid), LINKER_PARAM(sp_amp_linker,_sp_duid), COOKIE(_sp_duid)))"
     },
-    "_sp_id": {
-      "value": "LINKER_PARAM(linker, amp_id)"
+    "_sp_ampid": {
+      "value": "$IF(COOKIE(_sp_ampid), LINKER_PARAM(sp_amp_linker, amp_id), COOKIE(_sp_id))"
```

Comment:
I don't understand this line,

If _￼sp_ampid exists then use the linker param or if not use a different cookie￼? 
My guess is the cookie at the end is a mistake since you've deleted that above? I'm not sure why looking for that cookie means you'd want the linker param though?

Reply:
It is confusing!

So elsewhere, we call CLIENT_ID(sp_ampid) -> This macro will create a client ID if one doesn't exist, and will save it in the sp_ampid cookie. If the cookie does exist, it'll use that value. So if neither the linker param nor the _sp_id cookie exist, that macro will handle setting the cookie.

If the sp_ampid cookie doesn't exist, then either it should be a new client ID, OR this is a user who has a client ID, but we have changed the cookie name. So, if the new cookie doesn't exist, we attempt to set it to the value of the old cookie - in order not to change the Client ID for existing users in renaming the cookie.

If the sp_ampid cookie _does_ already exist, we use the linker param, purely because that's the existing behaviour. It was first implemented this way out of uncertainty as to whether we could rely purely on the cookie alone... In reality I think it's not needed, however since I can't set it to nothing I chose the previous behaviour. Like I said if the param isn't available it'll default to the CLIENT_ID() macro behaviour which sets it anyway.

I hope that makes more sense?

Explaining it this way already makes me feel like I may be missing something here - so if there's a simpler approach, or if we feel that it's best not to rename the cookie, I'm happy to reconsider.

Reply:
Yeah I think that makes more sense now.
Is `COOKIE(_sp_id)` expecting to find the JavaScript Tracker cookie? As I'm not sure that will since we append the domain hash to the name - or is this a cookie that is set by AMP using the parameters (which now I look closer I think it is but figured I'd ask for clarification)?

Reply:
The latter yeah. The hash being appended to the JS tracker cookie is why I didn't end up changing this before - but it just feels like a bad idea to name them the same.

As we discuss this, I am wondering if I've missed a specific edge case... If the _sp_id cookie exists on my own domain, and the user lands on the AMD CDN, creating a separate sp_ampid cookie, then there's a chance that the value is set to one thing on the CDN cookie, then reverts back to the old cookie value if they travel to my domain...

So I need to see if I can test. This might require a tweak.

Reply:
So, using the linker does have a purpose - when we move from one domain to another, this is how we preserve thew value. Commit incoming to establish the correct behaviour for my edge case.